http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/ClusterHelper.scala

```diff
@@ -112,13 +112,17 @@ class ClusterHelper(
 
           // build cluster configuration
           machineConfig = cluster.machineConfig
-          initScriptResources = if (dataprocConfig.customDataprocImage.isEmpty)
-            List(clusterResourcesConfig.initVmScript, clusterResourcesConfig.initActionsScript)
-          else List(clusterResourcesConfig.initActionsScript)
+          initScriptResources = List(clusterResourcesConfig.initActionsScript)
           initScripts = initScriptResources.map(resource => GcsPath(initBucketName, GcsObjectName(resource.value)))
           credentialsFileName = cluster.serviceAccountInfo.notebookServiceAccount
             .map(_ => s"/etc/${ClusterInitValues.serviceAccountCredentialsFilename}")
 
+          // If user is using https://github.com/DataBiosphere/terra-docker/tree/master#terra-base-images for jupyter image, then
+          // we will use the new custom dataproc image
+          dataprocImage = if (cluster.clusterImages.exists(_.imageUrl.contains("terra-jupyter")))
```

Comment:
Wondering if we should invert this?
```
if (cluster.clusterImages.exists(_.imageUrl == dataprocConfig.jupyterImage))
  dataprocConfig.legacyCustomDataprocImage
else dataprocConfig.customDataprocImage
```
That way the new one is the default, and we only use the legacy for the legacy image.

Reply:
I think that wouldn't work if somehow user uses an older legacy image than what's in config. But I agree this condition is not ideal...maybe this condition can be improved with your PR? (didn't mean you should do it. I'm happy to do a followup PR after yours now that your rstudio PR does the image detection in a more reliable way)

Reply:
True.. that's probably not very likely though